websites/H/Hyakanime/presence.ts

```diff
@@ -0,0 +1,56 @@
+const presence = new Presence({
+		clientId: "967174885992456292"
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "applogo_7fac0ec4359bda8ccf0f",
+			startTimestamp: browsingTimestamp
+		},
+		page = document.location.pathname;
+	if (page.startsWith("/anime/")) {
+		presenceData.details = "Regarde la fiche d'un animé:";
+		presenceData.state = document.querySelector(
+			"#root > div.anime-body > div.anime-white-informations > div.anime-white-infos > div:nth-child(2) > h1"
+		).textContent;
+		presenceData.buttons = [
+			{
+				label: "Fiche",
+				url: document.location.href
+			}
+		];
+	} else if (page.startsWith("/profil/")) {
+		presenceData.details = "Regarde le profil de:";
+		presenceData.state = document.querySelector(
+			"#root > div:nth-child(3) > div.profil-banner-top > div > div.bloc-name-banner > h1"
+		).textContent;
+		presenceData.buttons = [
+			{
+				label: "Profil",
+				url: document.location.href
+			}
+		];
+	} else if (page.includes("/search/"))
+		presenceData.details = "Cherche une fiche...";
+	else if (page.startsWith("/new"))
+		presenceData.details = "Soumets une nouvelle fiche";
+	else if (page.startsWith("/edit/")) presenceData.details = "Édit une fiche";
+	else if (page.startsWith("/bibliotheque"))
+		presenceData.details = "Regarde sa bibliothèque";
+	else if (page.startsWith("/agenda")) {
+		presenceData.details = "Regarde l'agenda";
+		presenceData.buttons = [
+			{
+				label: "Agenda",
+				url: document.location.href
+			}
+		];
+	}
+
+	if (presenceData.details) presence.setActivity(presenceData);
+	else {
+		presenceData.details = "Sur la page d'accueil";
```

Comment:
this should be set in the if chain above in the form of an else at the end

Reply:
I am not sure I understand, would it be possible to explain it to me again please